deploy-board/deploy_board/templates/hosts/hosts.tmpl

```diff
@@ -88,15 +88,15 @@
     <div class="pull-left">
         <button id="reset_{{ deployId }}"
                 class="deployToolTip btn btn-primary" data-toggle="tooltip"
-                title="Retry the current deploy step on this host.">
-            Retry deploy on all hosts
+                title="Retry deploys on all failed hosts.">
```

Comment:
Based on the query, it only retry current deploy
```java
    private static final String RESET_FAILED_AGENTS =
        "UPDATE agents SET state='RESET' " +
        "WHERE env_id=? AND deploy_id=? AND " +
        "status!='SUCCEEDED' AND status!='UNKNOWN' AND state!='RESET'";
```

Reply:
@tylerwowen yes, our intention is to restart deployment on all failed hosts not all deployments on the all failed hosts.

And this query will restart all failed hosts for current deployment. We don't want to restart all deployments on failed hosts, because there is a possibility that these failed hosts for current deployment will contains another not failed deployments, so it's not need to restart.

Reply:
My comment is regarding the change from "the current deploy step" to "deploys". It ls less clear after your change.